### ISSUE TITLE
Scrolling with the mouse has disappeared

### ISSUE THREAD

**Issue by MichelReims** [xournalpp/xournalpp#2523]
OS : Windows 10
Version of xournalpp : xournalpp-1.1.0+dev-nightly.20201210-windows.exe

Problem : It is no longer possible to scroll through the pages with the mouse wheel. This function was active in previous versions of xournalpp

**Comment by rolandlo** [comment]
I can't reproduce that on Linux.

**Comment by personalizedrefrigerator** [comment]
Do you have the `Touchscreen > Workaround` setting enabled? If so, does it work if you disable it?

**Comment by MichelReims** [comment]
Mouse wheel scrolling worked without this option in previous versions of xournalpp. I just put Workaround on "enabled, but It does not change anything ; It doesn't work

**Comment by MichelReims** [comment]
In "xournalpp-1.1.0+dev-nightly.20201203-windows" :  when I put  Workaround "enabled", the scrolling with mouse doesnt works
With Workaround "enabled" on my OS window 10),

**Comment by personalizedrefrigerator** [comment]
Thank you for checking!

**Comment by rolandlo** [comment]
I can't reproduce it on Windows either.
@MichelReims Try renaming your settings.xml file (after which a fresh one is created) and check again.

**Comment by MichelReims** [comment]
I just tried but it does not solve the problem

**Comment by MichelReims** [comment]
I just noticed that vertical scrolling with the mouse wheel works if you place the mouse pointer on the vertical scroll bar and use the mouse wheel; if the pointer is not on the vertical scroll bar, the scroll window does not work with the mouse wheel

**Comment by personalizedrefrigerator** [comment]
@MichelReims If you have time (and know your way around git), would you be willing to find the commit that introduced this regression? If so, this would be very helpful!

**Comment by MichelReims** [comment]
I know that this problem did not exist in the nightly release of the 2020-12-03. This problem appeared after this version

**Comment by personalizedrefrigerator** [comment]
Do you know if the problem is present in 49cf45ccf646b142f0beec8c9e8a0cbaf323b57a? 666bf465c7070c4d720c50239bd7b3d7c1c594d0? e43ea4100e59a0f81f253a6348cfa776540d8b78? a57727375c06a390f4888644ad1f2c99fc0737ad?
I understand that building from source can take a long time, but I can't reproduce the issue. As such, it would be very helpful to know which commit caused the regression.

Thank you!

**Comment by MichelReims** [comment]
Yes i will compile and test.

**Comment by MichelReims** [comment]
I updated pacman and recompiled versions  49cf45c , 666bf46 , e43ea41  and  a577273 .Scrolling with the mouse wheel works

**Comment by MichelReims** [comment]
What other version do you want I test ?

**Comment by personalizedrefrigerator** [comment]
de3b57e34f6a1ea83b5405190e3b7820838dad2a, a19f79170d233c3a062177cc604630188e0f3e74

**Comment by MichelReims** [comment]
Ok

**Comment by personalizedrefrigerator** [comment]
If it fails with both of those, try b47cf4d110011ac2fe9624994b0fc5f469953df8. If it works, try 6a6fddde7afd1640af958e7a8e2faadd96d720af.

**Comment by MichelReims** [comment]
I test de3b57e, a19f791 and b47cf4d . All fails : Scrolling with the mouse wheel doesn't work.

**Comment by personalizedrefrigerator** [comment]
Okay, so it's between https://github.com/xournalpp/xournalpp/commit/b47cf4d110011ac2fe9624994b0fc5f469953df8 and 666bf465c7070c4d720c50239bd7b3d7c1c594d0...

Could you try bd0b5bb717351c9c2a9462ad6c2e4778db99c9ed, d8f393eb831225a5fed59797cb708a3cba13cf5d, fc41aafe5fa535838f3d1e50e43dfdd956ed4aa5, and 925acf9eefce1775bb3b174d2742d620243a3412? (At this point, I'm just looking through the commit history between b47cf4d110011ac2fe9624994b0fc5f469953df8 and 666bf465c7070c4d720c50239bd7b3d7c1c594d0... feel free to try other commits)

**Comment by MichelReims** [comment]
Ok I will test

**Comment by MichelReims** [comment]
Sorry. All tests fails : Scrolling with the mouse wheel doesn't work. I don't understand

**Comment by personalizedrefrigerator** [comment]
My guess is it's commit 925acf9eefce1775bb3b174d2742d620243a3412!!! If you want to double-check, the commit just before this is 891c972c27a335113f8a424ae7d3adcbc1ed3f18.

**Comment by personalizedrefrigerator** [comment]
@Febbe Could you take a look at this?

**Comment by MichelReims** [comment]
I find : the problem is in commit  925acf9eefce1775bb3b174d2742d620243a3412 of 2020-12-09 ("Initialize xournalpp over GtkApplication ")
Commit 891c972c27a335113f8a424ae7d3adcbc1ed3f18 works with wheel of mouse but not commit  925acf9eefce1775bb3b174d2742d620243a3412

**Comment by Febbe** [comment]
I can't imagine that I touched the input system neither on gtk/glib's side neither on xournalpp's side. I'll try to reproduce this on windows.

**Comment by Febbe** [comment]
I managed to reproduce this with commit 891c972:
It is caused by the `GTK Touch / Scrolling workaround`. It is buggy and should be removed soon.

**Comment by MichelReims** [comment]
Yes it"s exactly the same issue  under windows 10

**Comment by Febbe** [comment]
The thing is, that this issue was definitely there before, so it's not new. I think you are experience those, because there either were settings before in 'C:\Users\<YourUserName>\AppData\Local\xournalpp` or the settings have been reset because the migration was buggy. (Did not find 'C:\Users\<YourUserName>\AppData\Local\com.github.xournalpp.xournalpp`)

**Comment by MichelReims** [comment]
How can I solve the problem ?

**Comment by Febbe** [comment]
Either delete all your setting files. Located in `C:\Users\<YourUserName>\AppData\Local\xournalpp` , `C:\Users\<YourUserName>\AppData\Local\com.github.xournalpp.xournalpp` and `C:\Users\<YourUserName>\.xournalpp` . 

Or go to `Settings > Touchscreen > Workaround` and disable the workaround

**Comment by MichelReims** [comment]
This does not solve the problem.
When my mouse is on the vertical scroll bar, vertical scrolling works with the mouse wheel; when my mouse is inside the window the scrolling no longer works with the mouse wheel. It is the same symptom as #2540

**Comment by MichelReims** [comment]
I noticed that the "C: \ Users \ <YourUserName> \ AppData \ Local \ Gtk-3.0" folder is empty. I don't know if it's normal

**Comment by Febbe** [comment]
So you mean, that it also disappeared with the touch workaround disabled?

**Comment by MichelReims** [comment]
Yes, with workaround disabled or enabled, the vertical scrolling with the wheel of mouse doesn't work when the pointer of mouse is  not on vertical scrolbar

**Comment by MichelReims** [comment]
To be sure that my problem does not come from a previous version of xournalpp which would have left incompatible files, I installed "xournalpp-1.1.0+dev-nightly.20201216-windows" on a windows 10 on which xournalpp had never been installed; it doesn't change anything; the problem persists

**Comment by Febbe** [comment]
OK can you send me your settingsfolder. Also you could try to compile with INPUT_DEBUG. But to help you with that, i need to study it my self. @LittleHubba can help you tho.

**Comment by MichelReims** [comment]
I found something : i have reinstalled MSYS2, NSIS then compiled last commit b5b42d6 ; the problem disappeared with the version that I compiled
But when I install the commit 67b55ed17ca1d78a99637301eaa075f6a63c68da ("xournalpp-1.1.0+dev-nightly.20201220-windows") the problem reappears

**Comment by Febbe** [comment]
> 
> 
> But when I install "xournalpp-1.1.0+dev-nightly.20201220-windows" (commit [67b55ed](https://github.com/xournalpp/xournalpp/commit/67b55ed17ca1d78a99637301eaa075f6a63c68da)) the problem reappears on 2 computers (windows 7 and 10)
> I don't know why it's solved when I compile the source

Thank you very much for that hint. I can reproduce it now.
On a first look, it seems like, that one of the libraries is the issue.

**Comment by Febbe** [comment]
Compiling with MinSizeRel will break it. Somehow switching from RelWithDebInfo, broke nightly builds.

**Comment by MichelReims** [comment]
I didn't know how to program so I don't understand what you mean.  The most important thing is that you saw where the problem came from

**Comment by Febbe** [comment]
The pipeline should work tonight, but the issue itself is not fixed. 
It's also very hard to find the cause. We just don't use `MinSizeRel` anymore to build. 
So don't wounder, why I marked this with low priority.

**Comment by ErgastiAlex** [comment]
I have the same bug on windows as reported in #2588 but i just downloaded the version on the last Nightly Build.
There is a workaround, luckily, because this bug is pretty annoying and makes the app unusable.
I created this plugin that map 2 key to scroll up and scroll down, i used to have Ctrl + and Ctrl - on 2 key of my graphics tablet, now i just need to remap this two key to the new combination.

function initUi()
	app.registerUi({["menu"]="Scroll Down", ["callback"]="scroll_down",["accelerator"]="<ctrl><alt>D"});
	app.registerUi({["menu"]="Scroll Up", ["callback"]="scroll_up",["accelerator"]="<ctrl><alt>U"});
end

function scroll_down()
	app.scrollToPos(0,10);
end

function scroll_up()
	app.scrollToPos(0,-10);
end
